src/esp/gfx/RenderingTarget.cpp

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <Magnum/GL/Buffer.h>
+#include <Magnum/GL/DefaultFramebuffer.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/TextureFormat.h>
+#include <Magnum/Image.h>
+#include <Magnum/PixelFormat.h>
+
+#include "RenderingTarget.h"
+#include "magnum.h"
+
+using namespace Magnum;
+
+namespace esp {
+namespace gfx {
+struct RenderingTarget::Impl {
+  Impl(WindowlessContext::ptr context, int height, int width)
+      : context_{context},
+        framebufferSize_(width, height),
+        colorBuffer_(),
+        depthBuffer_(),
+        objectIdBuffer_(),
+        depthRenderbuffer_(),
+        framebuffer_(Magnum::NoCreate) {
+    colorBuffer_.setStorage(GL::RenderbufferFormat::SRGB8Alpha8,
+                            framebufferSize_);
+    depthBuffer_.setStorage(GL::RenderbufferFormat::R32F, framebufferSize_);
+    objectIdBuffer_.setStorage(GL::RenderbufferFormat::R32UI, framebufferSize_);
+    depthRenderbuffer_.setStorage(GL::RenderbufferFormat::Depth24Stencil8,
+                                  framebufferSize_);
+
+    framebuffer_ = GL::Framebuffer{{{}, framebufferSize_}};
+    framebuffer_
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{0}, colorBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{1}, depthBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{2},
+                            objectIdBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::BufferAttachment::Depth,
+                            depthRenderbuffer_)
+        .mapForDraw({{0, GL::Framebuffer::ColorAttachment{0}},
+                     {1, GL::Framebuffer::ColorAttachment{1}},
+                     {2, GL::Framebuffer::ColorAttachment{2}}});
+    CORRADE_INTERNAL_ASSERT(
+        framebuffer_.checkStatus(GL::FramebufferTarget::Draw) ==
+        GL::Framebuffer::Status::Complete);
+  }
+
+  void renderEnter() {
+    framebuffer_.bind();
+    framebuffer_.clear(GL::FramebufferClear::Color |
+                       GL::FramebufferClear::Depth);
+    framebuffer_.clearColor(1, Vector4{});
+    framebuffer_.clearColor(2, Vector4ui{});
+    framebuffer_.bind();
+  }
+
+  void renderExit() { GL::defaultFramebuffer.bind(); }
+
+  gltensor::GLTensorParam::ptr glTensorParam() const {
+    auto param = std::make_shared<gltensor::GLTensorParam>();
+
+    param->height_ = framebufferSize_[1];
```

Comment:
You may consider:
define the following two in the core/esp.h and use them everywhere in the code.
const int WIDTH = 0;
const int HEIGHT = 1;

Reply:
It differs if we are using graphics or computer vision conventions.  Since we interact with both all the time, it's gonna differ all the time.  i.e the numpy arrays and GL Tensor (which is targeting CV people) have things in h x w format.

Reply:
Why not just `framebufferSize.y()`? No constants needed.